SU-SWS/decanter
# Add parameter to @link-icon to adjust vertical position of icon and different options for $animate parameter

# READY FOR REVIEW

# Summary
- Add a parameter `$vertical` in the `@link-icon` mixin so users can adjust the vertical position of the icon by adding a corresponding `margin-bottom` to the `::after` pseudo element of the link. Positive value moves icon up; negative moves it down.
- Add options for `$animate: right, up, down, topright` (diagonally towards upper right corner).
- Finetune icon vertical positions of existing link component variants.

# Needed By (Date)
- Whenever

# Urgency
- Not urgent

# Steps to Test

1. Pull this branch and compile styleguide.
2. Go to the compiled styleguide, look at Components -> Links, check that all the icons of existing link variants have good vertical alignments with the link text and that they animate as specified in #505 
3. Open a link variant scss file, e.g., components/link/_link--internal.scss, change the `$vertical `and `$animate` parameters and check that they behave as expected.

# Affected Projects or Products
- Decanter and any sites that use link variants that uses SVG masks

# Associated Issues and/or People
- #456 
- #505
